### ISSUE TITLE
Strange errors about missing include files that aren't missing

### ISSUE THREAD

**Issue by paxcut** [Viladoman/StructLayout#22]
As a simple example I am trying your Visual Studio TestProject sample. I can layout Base from TestHeader.h without errors, but if I try TestStruct from TestProject.cpp I get error:

[04:57:01] Using manual configuration...
[04:57:01] Looking for structures at H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp:7:16...
[04:57:01] COMMAND LINE: H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp -- -x c++ -m64
[04:57:01] Execution Log:
H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp:1:10: fatal error: 'cstdio' file not found
#include <cstdio>
         ^~~~~~~~
1 error generated.
Error while processing H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp.

[04:57:01] [ERROR] Unable to scan the given location. (313.0954 ms)

but when I compile the project there are no errors and I can open cstdio without a problem. 

If I force include cstdio then it complains about another header and if that one is forced then another and another ad infinitum. Has anybody ran into a similar situation? The truth is that I am getting a similar error in a more complex project but the symptoms are the same. 

I have tried other options, and visual studio settings but nothing helps. None of the current issues seem similar (at least that's the impression I get and i have read them all). Does anybody have any suggestions or things I may try to find the source of my problem? Anything would be greatly appreciated as this extension can be very useful but I am totally stumped. 

I realize that the problem is with my system as nobody else seems to talk about it but other than this extension I have no problems with visual studio (2019). 
Thank you.

Paxcut

**Comment by Viladoman** [comment]
Hi, thanks for letting me know about this! 

From the logs I am seeing 2 different things here: 
- `Using manual configuration...` this means, unless there is an internal issue, that the extension option for **Automatic extraction** is disabled. (for the example test, it will make the .cpp fail to find the testheader.h). Still this won't fix the main issue.
- Could you add `-v` to the **Extra Args** in the extension options so it prints the include paths. This way we can see where it is trying to get the STL from. ( you can also try `-H` for a more detailed header usage)

it would also be nice to know the Visual Studio version you are running on. 

Let me know how it goes.

**Comment by paxcut** [comment]
sorry about the manual configuration. Out of desperation I tried to change options and forgot to put them back but it is fixed now. Visual Studio is 2019 Enterprise 16.8.3. The cstdio file is located at C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\include
 here is output after adding -v I don't see the include directory in it, but project compiles. thank you for the assistance.

[12:06:24] Capturing configuration from VS projects...
[12:06:24] Looking for structures at H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp:7:14...
[12:06:24] COMMAND LINE: H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp -- -x c++ -m32 -std=c++17 -w -DWIN32 -DNDEBUG -D_CONSOLE -I"Folder Space" -ID:\temp\vcpkg\installed\x86-windows\include -working-directory=H:\temp\StructLayout\TestProjects\Regular -v
[12:06:24] Execution Log:
clang version 12.0.0 (https://github.com/llvm/llvm-project.git 9087209314caafed4b232d4a66287f2d16054ad3)
Target: i686-pc-windows-msvc
Thread model: posix
InstalledDir: 
clang version 12.0.0 (https://github.com/llvm/llvm-project.git 9087209314caafed4b232d4a66287f2d16054ad3)
Target: i686-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE
clang Invocation:
 "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\IDE\\clang-tool" "-cc1" "-triple" "i686-pc-windows-msvc19.28.29335" "-fsyntax-only" "-disable-free" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "TestProject.cpp" "-mrelocation-model" "static" "-mframe-pointer=all" "-fmath-errno" "-fno-rounding-math" "-mconstructor-aliases" "-target-cpu" "pentium4" "-tune-cpu" "generic" "-v" "-resource-dir" "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\lib\\clang\\12.0.0" "-working-directory" "H:\\temp\\StructLayout\\TestProjects\\Regular" "-D" "WIN32" "-D" "NDEBUG" "-D" "_CONSOLE" "-I" "Folder Space" "-I" "D:\\temp\\vcpkg\\installed\\x86-windows\\include" "-internal-isystem" "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\lib\\clang\\12.0.0\\include" "-internal-isystem" "c:\\program files (x86)\\devstudio\\vc\\include" "-internal-isystem" "c:\\program files (x86)\\devstudio\\vc\\atl\\include" "-internal-isystem" "c:\\program files (x86)\\devstudio\\vc\\mfc\\include" "-internal-isystem" "c:\\program files (x86)\\microsoft visual studio .net\\frameworksdk\\include\\" "-internal-isystem" "C:\\Program Files (x86)\\Microsoft Visual Studio .NET\\FrameworkSDK\\include\\" "-w" "-std=c++17" "-fdeprecated-macro" "-fdebug-compilation-dir" "H:\\temp\\StructLayout\\TestProjects\\Regular" "-ferror-limit" "19" "-fno-use-cxa-atexit" "-fms-extensions" "-fms-compatibility" "-fms-compatibility-version=19.28.29335" "-fdelayed-template-parsing" "-fcxx-exceptions" "-fexceptions" "-faddrsig" "-x" "c++" "H:\\temp\\StructLayout\\TestProjects\\Regular\\TestProject.cpp"

clang -cc1 version 12.0.0 based upon LLVM 12.0.0git default target i686-pc-windows-msvc
ignoring nonexistent directory "D:\temp\vcpkg\installed\x86-windows\include"
ignoring nonexistent directory "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\lib\clang\12.0.0\include"
ignoring nonexistent directory "c:\program files (x86)\devstudio\vc\include"
ignoring nonexistent directory "c:\program files (x86)\devstudio\vc\atl\include"
ignoring nonexistent directory "c:\program files (x86)\devstudio\vc\mfc\include"
ignoring duplicate directory "c:\program files (x86)\microsoft visual studio .net\frameworksdk\include"
#include "..." search starts here:
#include <...> search starts here:
 Folder Space
 c:\program files (x86)\microsoft visual studio .net\frameworksdk\include
End of search list.
H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp:1:10: fatal error: 'cstdio' file not found
#include <cstdio>
         ^~~~~~~~
1 error generated.
Error while processing H:\temp\StructLayout\TestProjects\Regular\TestProject.cpp.

[12:06:24] [ERROR] Unable to scan the given location. (41.0324 ms)

**Comment by Viladoman** [comment]
It looks like Clang in your windows setup thinks that the MSVC toolchain is under `c:\program files (x86)\devstudio\vc`

My example information looks more like this: 
```
ignoring nonexistent directory "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\lib\clang\12.0.0\include"
#include "..." search starts here:
#include <...> search starts here:
 Folder Space
 C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\include
 C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\atlmfc\include
 C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\ucrt
 C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared
 C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um
 C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt
End of search list.
```

Can you please try adding them to the clang system path with `-isystem"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\include"` on the **Extra Args**? Ideally the whole list above (with your equivalent paths).

I am not super familiar on how this clang toolchain detection works. 

Found this related [stackoverflow issue](https://stackoverflow.com/questions/42256526/clang-on-windows-installed-via-scoop-is-unable-to-find-headers). It might bring some leads on how to make clang find the MS STL path by default.

**Comment by paxcut** [comment]
I added the changes you suggested but wasn't sure how to add multiple directories. I tried a colon separated long string but that didn't work. What worked was a series of space separated -isystem switches each with its own directory. 

That got rid of the missing include file error but then I ran into the intrin0.h error due to my Visual Studio version. I applied the diff as the current workaround suggests and now it works flawlessly. 

I'll look at the stackoverflow article to try to understand why my clang installation has such misplaced notion of where MSVC toolchain is located. I will make appropriate changes to the other project I mentioned but I have no doubt that it will work out as well.

I have used clang a little in the past as it was required for some projects I worked on  but I am very unfamiliar with its inner workings, specially its relation to Visual Studio. I am really impressed by how quickly you responded and resolved this problem. 
As the problem reported has been fixed I think I'll close this issue now. Thanks again and good luck in your future endeavors. Congrats on the very useful extension you created.
